src/test/java/com/google/devtools/common/options/ExpressionConverterTest.java

```diff
@@ -0,0 +1,56 @@
+package com.google.devtools.common.options;
+
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.devtools.common.options.Converters.ExpressionConverter;
+import static org.junit.Assert.assertThrows;
+
+import com.google.devtools.build.lib.syntax.Expression;
+import com.google.devtools.build.lib.syntax.ParserInput;
+import com.google.devtools.build.lib.syntax.SyntaxError;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** A test for {@link ExpressionConverter}. */
+@RunWith(JUnit4.class)
+public class ExpressionConverterTest {
```

Comment:
Don't test ExpressionConverter: it is not a public API. Instead, test its effects through the blaze test command, as that is the interface that actually matters.

Reply:
Should I do this by adding a shell script test in [src/test/shell/integration](https://github.com/bazelbuild/bazel/tree/master/src/test/shell/integration)?

Reply:
> Making this class private will not work since the methods **convert** and **getTypeDescription** must be accessible outside of **LoadingOptions**. Do you still want it to be moved but stay public or is it better if it stays in **Converters**?

If this class is public should we keep the tests as they are?